src/test/java/teammates/test/pageobjects/AppPage.java

```diff
@@ -837,13 +838,24 @@ public AppPage verifyHtmlPart(By by, String filePath) {
         }
         return this;
     }
-
-    private String extractHtmlPartFromFile(By by, String filePath)
+    
+    private String extractHtmlPartFromPageSource(By by, String pageSource) throws SAXException, IOException {
+        InputSource inputSource = new InputSource();
+        inputSource.setCharacterStream(new StringReader(pageSource));
+        return extractHtmlPartFromInputSource(by, inputSource);
+    }
+    
+    private String extractHtmlPartFromFile(By by, String filePath) throws SAXException, IOException {
+        InputSource inputSource = new InputSource(new BufferedReader(new FileReader(filePath)));
+        return extractHtmlPartFromInputSource(by, inputSource);
+    }
+    
+    private String extractHtmlPartFromInputSource(By by, InputSource inputSource)
             throws SAXException, IOException {
         String byId = by.toString().split(":")[1].trim();
         
         DOMParser parser = new DOMParser();
-        parser.parse(new InputSource(new BufferedReader(new FileReader(filePath))));
+        parser.parse(inputSource);
         org.w3c.dom.Document htmlDoc = parser.getDocument();
         org.w3c.dom.Element expectedElement = htmlDoc.getElementById(byId);
```

Comment:
Not relevant to this PR, but you may want to extract the package name up here because we don't usually specify the full package name when instantiating an object.
Or you may want it to open as a first-timer issue.